src/Forms/SearchableDropdownTrait.php

```diff
@@ -0,0 +1,594 @@
+     * will be remvoved in a future version
+     */
+    public function setPlaceholder(string $placeholder): static
+    {
+        $this->placeholder = $placeholder;
+        // SearchableDropdownField will have the setHasEmptyDefault() method from SingleSelectField
+        if (method_exists($this, 'setHasEmptyDefault')) {
+            $this->setHasEmptyDefault(true);
+        }
+        return $this;
+    }
+
+    /**
+     * Get the search context to use
+     * If a search context has been set via setSearchContext() that will be used
+     * Will fallback to using the dataobjects default search context if a sourceList has been set
+     * Otherwise will return null
+     */
+    public function getSearchContext(): ?SearchContext
+    {
+        if ($this->searchContext) {
+            return $this->searchContext;
+        }
+        if ($this->sourceList) {
+            $dataClass = $this->sourceList->dataClass();
+            /** @var DataObject $obj */
+            $obj = $dataClass::create();
+            return $obj->getDefaultSearchContext();
+        }
+        return null;
+    }
+
+    /**
+     * Set the search context to use instead of the dataobjects default search context
+     *
+     * Calling this will also call setUseSearchContext(true)
+     */
+    public function setSearchContext(?SearchContext $searchContext): static
+    {
+        $this->searchContext = $searchContext;
+        $this->setUseSearchContext(true);
+        return $this;
+    }
+
+    /**
+     * Get whether to use a dynamic placeholder if a normal placeholder is not set
+     */
+    public function getUseDynamicPlaceholder(): bool
+    {
+        return $this->useDynamicPlaceholder;
+    }
+
+    /**
+     * Set whether to use a dynamic placeholder if a normal placeholder is not set
+     */
+    public function setUseDynamicPlaceholder(bool $useDynamicPlaceholder): static
+    {
+        $this->useDynamicPlaceholder = $useDynamicPlaceholder;
+        return $this;
+    }
+
+    /**
+     * Get whether to use a search context instead searching on labelField
+     */
+    public function getUseSearchContext(): bool
+    {
+        return $this->useSearchContext;
+    }
+
+    /**
+     * Set whether to use a search context instead searching on labelField
+     */
+    public function setUseSearchContext(bool $useSearchContext): static
+    {
+        $this->useSearchContext = $useSearchContext;
+        return $this;
+    }
+
+    /**
+     * Get whether the field allows searching by typing characters into field
+     */
+    public function getIsSearchable(): bool
+    {
+        return $this->isSearchable;
+    }
+
+    /**
+     * Set whether the field allows searching by typing characters into field
+     */
+    public function setIsSearchable(bool $isSearchable): static
+    {
+        $this->isSearchable = $isSearchable;
+        return $this;
+    }
```

Comment:
There are use cases for very simply drop downs that only have say 3 options and searching doesn't really make sense.  Maybe it's not the greatest example since it wouldn't use this component though something like the mode selector

![image](https://github.com/silverstripe/silverstripe-framework/assets/4809037/4e0c82b8-81d5-463a-9654-9b4a0e80002d)

Reply:
I'd say that those use cases just shouldn't use this field - they should use `DropdownField` directly instead, or `ListboxField` for selecting multiple options. This field is explicitly `SearchableDropdownField` - we shouldn't allow it to not be searchable.

Reply:
Ideal plan is in a CMS major get rid of Dropdownfield and ListboxField (which use jquery) and and turn SearchableDropdownFieldTrait in a new Dropdownfield which does all of the things.  If we do that then we'll want the flexibility to have in non-searchable.

Reply:
I really think that should be done at the time we want to make that change - it's completely nonsensical for it to be not searchable _now_, regardless of future plans.

That said, ultimately having a method that nobody calls because it's a silly method won't hurt anything, it's just untidy. So for the sake of getting things done I'll drop this.